src/RolloutScorer/RolloutScorer.Service/Program.cs

```diff
@@ -0,0 +1,153 @@
+            services.Configure<GitHubClientOptions>(
+                options => { options.ProductHeader = new Octokit.ProductHeaderValue(assemblyName, assemblyVersion); }
+            );
+            services.Configure<GitHubTokenProviderOptions>("GitHubAppAuth", (o, s) => s.Bind(o));
+            services.Configure<AzureDevOpsSettings>("AzureDevOps", (o, c) => c.Bind(o));
+            services.Configure<GitHubSettings>("GitHub", (o, c) => c.Bind(o));
+        }
+
+        public static void AddServices(IServiceCollection services)
+        {
+            services.AddSingleton<IScorecardService, ScorecardProvider>();
+            services.AddSingleton<IRolloutScorerService, RolloutScorerProvider>();
+            services.AddSingleton<IIssueService, IssueProvider>();
+
+            services.AddSingleton<IGitHubClientFactory, GitHubClientFactory>();
+            services.AddSingleton<IGitHubApplicationClientFactory, GitHubApplicationClientFactory>();
+            services.AddGitHubTokenProvider();
+            services.AddSingleton<IAzureDevOpsClientFactory, AzureDevOpsClientFactory>();
+
+            (string assemblyName, string assemblyVersion) = GetAssemblyVersion();
+
+            services.AddSingleton<IGitHubClient>(
+                provider =>
+                {
+                    var options = provider.GetRequiredService<IOptions<GitHubSettings>>().Value;
+                    var clientFactory = provider.GetRequiredService<IGitHubApplicationClientFactory>();
+                    return clientFactory.CreateGitHubClientAsync(options.Organization, options.Repository).Result;
+                });
+
+            services.AddSingleton<VssCredentials>(
+                provider =>
+                {
+                    var options = provider.GetRequiredService<IOptions<AzureDevOpsSettings>>();
+                    return new VssBasicCredential("", options.Value.AccessToken);
+                }
+            );
+
+            services.AddSingleton<VssConnection>(
+                p =>
+                {
+                    var options = p.GetRequiredService<IOptions<AzureDevOpsSettings>>().Value;
+                    var credentials = p.GetRequiredService<VssCredentials>();
+                    var settings = new VssClientHttpRequestSettings
+                    {
+                        UserAgent = new List<ProductInfoHeaderValue>
+                            {new ProductInfoHeaderValue(assemblyName, assemblyVersion)}
+                    };
+                    return new VssConnection(
+                        new Uri(options.CollectionUri, UriKind.Absolute),
+                        new VssHttpMessageHandler(credentials, settings),
+                        p.GetServices<AzureDevOpsDelegatingHandler>()
+                    );
+                }
+            );
+
+            services.TryAddSingleton(p => p.GetRequiredService<VssConnection>().GetClient<BuildHttpClient>());
+
+
+            //services.AddAzureStorageQueue();
```

Comment:
Copied from Build Analysis project and should be cleaned up. (Wanted to use some of the same DI we used to interface with AzDO and GitHub).